File: frontend/src/app/models/producto.ts
import { Unidad } from "./unidad";

export interface Producto {
    id: number;
    nombre: string;
    descripcion: string;
    codigoBarras: string;
    precioCompra : number;
    precioVenta: number;
    cantidad: number;
    unidad: Unidad;
}

export interface ProductoTable{
    producto: Producto;
    cantidad: number;
    subtotal: number;
}

File: backend/src/main/java/mytiendita/backend/model/Producto.java
package mytiendita.backend.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "producto")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "descripcion")
    private String descripcion;

    @Column(name = "codigo_barras")
    private String codigoBarras;

    @Column(name = "precio_compra")
    private Double precioCompra;

    @Column(name = "precio_venta")
    private Double precioVenta;

    @Column(name = "cantidad")
    private Double cantidad;

    @ManyToOne
    @JoinColumn(name = "id_unidad", referencedColumnName = "id")
    private Unidad unidad;
}
